pkg/messaging/direct_messaging.go

```diff
@@ -77,45 +75,47 @@ type remoteApp struct {
 	address   string
 }
 
+// NewDirectMessaging contains the options for NewDirectMessaging.
+type NewDirectMessagingOpts struct {
+	AppID               string
+	Namespace           string
+	Port                int
+	Mode                modes.DaprMode
+	AppChannel          channel.AppChannel
+	ClientConnFn        messageClientConnection
+	Resolver            nr.Resolver
+	MaxRequestBodySize  int
+	Proxy               Proxy
+	ReadBufferSize      int
+	Resiliency          resiliency.Provider
+	IsResiliencyEnabled bool
+}
+
 // NewDirectMessaging returns a new direct messaging api.
-func NewDirectMessaging(
-	appID, namespace string,
-	port int, mode modes.DaprMode,
-	appChannel channel.AppChannel,
-	clientConnFn messageClientConnection,
-	resolver nr.Resolver,
-	tracingSpec config.TracingSpec,
-	maxRequestBodySize int,
-	proxy Proxy,
-	readBufferSize int,
-	streamRequestBody bool,
-	resiliency resiliency.Provider,
-	isResiliencyEnabled bool,
-) DirectMessaging {
+func NewDirectMessaging(opts NewDirectMessagingOpts) DirectMessaging {
```

Comment:
Why this change: the old method's signature had 15 arguments and it was very hard to maintain when we needed to remove or add parameters (like in this case), especially in tests

Reply:
There are some slight performance difference in using the options pattern vs the named params. But I don't recall which one is better haha

Reply:
This may be slightly worse, but this code path is only executed once during initialization.